### PR TITLE
add infer analysis documentation and reports.

### PR DESCRIPTION
**Pull request by Tacinight**

### CODE REVIEW

File: infer/Setup.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# make install
+```
+
+6. To run infer on Linux kernel code, there are several dirctories needed to skip. Under the root directory of Linu kernel code, add a new file named `.inferconfig`. Add those dircotries for stable v4.15, "arch/x86", "arch", "mm", "drivers/acpi", "fs", "kernel/bpf", "net/mac80211". In my case, the file looks like:
+ 
+```
+{
+    "skip-analysis-in-path": ["home/evan/Repository/linux/stable/linux-stable/arch/x86", "home/evan/Repository/linux/stable/linux-stable/arch", "home/evan/Repository/linux/stable/linux-stable/mm", "home/evan/Repository/linux/stable/linux-stable/drivers/acpi", "home/evan/Repository/linux/stable/linux-stable/fs", "home/evan/Repository/linux/stable/linux-stable/kernel/bpf", "home/evan/Repository/linux/stable/linux-stable/net/mac80211"]
+}
+```
```
Comment (bulwahn) [inline infer/Setup.md]: What do you execute after this setup to obtain your results?

Comment (Tacinight) [reply infer/Setup.md]: Just run 'infer run -- make -jn' under the root directory of Linux kernel code.
I will update this later.

Comment (bulwahn) [reply infer/Setup.md]: Yes, but what is the .config you are using?

**Comment by bulwahn** [comment]
I would propose the following directory structure under infer:
- directory Documentation for your installation instructions
- directory Reports for your manual analysis reports
  - directory Reports/images/ for images in the reports
- directory models: for the infer model files (but it might be better to have them in your fork of the infer repository and just refer to them via URL.)
- directory results:
  - subdirectories v1, v2, v3, v4, just an increasing number for now. I believe it is difficult to say what a minor and major version is, so it might be simpler to just increase numbers.
- directory scripts: basic scripts to run certain tasks (no content here now, but certainly later).

**Comment by bulwahn** [comment]
I would like to have the raw results as reference for a reproducible process.
So, my idea is:
there is a bug.txt in results/v1, and there is all information to reproduce this file, e.g., which kernel version, which kernel config, which infer version, which infer model, in some simple config file in results/v1.
Then there is a script in scripts and we can run it, e.g. with `./rerun-infer.sh v1`. It then sets up everything according to the config file and run infer and produces an own local `bugs.txt` file. Then it compares to the `bugs.txt` file in the repository and it should always equal to that. This will simplify our discussion because we know that we have the same setup and we can quickly confirm this. This would be my thought on a final setup, let us start moving into that direction.

File: infer/Setup.md (hunk 1 above)
Comment (Tacinight) [reply infer/Setup.md]: `make defconfig`

**Comment by Tacinight** [comment]
That sounds reasonable, if there is no other suggestion, I will close this pl, fix the problems and pull another pl.

**Comment by bulwahn** [comment]
You don't need to close this pull request (and you should not), you can update your local branch on your local machine, and then push (with force) to your branch on your github repository and that will update this pull request.

**Comment by Tacinight** [comment]
I was thinking to rebase/reset the log history since some of them are just overridden. Force push would be OK.

**Comment by bulwahn** [comment]
Sure you can do that; but you don't need to close this pull request, you can just update it and I will continue reviewing it.

**Comment by Tacinight** [comment]
I feel it's unnecessary to split them into different commits, so I just group them together.

**Comment by bulwahn** [comment]
Sure, you can do that it is not that large that I could not review it anymore.

**Comment by bulwahn** [comment]
The content of the pull request looks good to me. I will merge it.

**Pull request merged**